### PR TITLE
Add Athena Workgroup config

### PR DESCRIPTION
**Pull request by ndrluis**

### CODE REVIEW

File: target_athena/athena.py [hunk 1]
```diff
@@ -30,6 +30,9 @@ def create_client(config, logger: Logger):
     aws_profile = config.get("aws_profile") or os.environ.get("AWS_PROFILE")
     aws_region = config.get("aws_region") or os.environ.get("AWS_REGION")
     s3_staging_dir = config.get("s3_staging_dir") or os.environ.get("S3_STAGING_DIR")
+    athena_workgroup = config.get("athena_workgroup") or os.environ.get(
+        "ATHENA_WORKGROUP"
+    )
```
Comment (aaronsteers) [inline target_athena/athena.py:35]: Is it okay to remove the coalesce to the environment variable?

If `--config=ENV` is specified at the command line, the same can be set with the convention `<connector_name>_<setting_name>`, as in `TARGET_ATHENA_ATHENA_WORKGROUP`.

(The `--config=ENV` arg can be used in combination with one or more `config=path/to/config.json` args.)

More info: https://sdk.meltano.com/en/latest/implementation/cli.html#parsing-environment-variables

**Review by aaronsteers — CHANGES_REQUESTED** [review]
Just one requested change.

File: target_athena/target.py [hunk 2]
```diff
@@ -33,6 +33,7 @@ class TargetAthena(Target):
         th.Property("temp_dir", th.StringType),
         th.Property("stream_maps", th.ObjectType()),
         th.Property("stream_map_config", th.ObjectType()),
+        th.Property("athena_workgroup", th.StringType, default="primary"),
```
Comment (aaronsteers) [inline target_athena/target.py:36]: What do you think of `athena_workgroup` vs `workgroup` vs `workgroup_name`?

I have no problem with `athena_workgroup` but wanted to see if you have any thoughts on removing the `athena_` prefix or if you considered any possible alternates.

File: target_athena/athena.py (hunk 1 above)
Comment (ndrluis) [reply target_athena/athena.py]: Am I using the --config option with env var incorrectly?
image

**Comment by giamo** [comment]
Any chance this will be merged? It's useful when working with multiple workgroups

**Comment by tayloramurphy** [comment]
@pnadolny13 @edgarrmondragon thoughts about getting this across the finish line?

**Comment by edgarrmondragon** [comment]
> @pnadolny13 @edgarrmondragon thoughts about getting this across the finish line?

@tayloramurphy Seems easy enough to apply AJ's suggestions, though I don't know if we have an Athena deployment with which to test the changes

**Comment by pnadolny13** [comment]
@edgarrmondragon @tayloramurphy I was able to test this on a personal athena database and it worked as expected. I actually dont think we need to apply any changes, AJ made some minor suggestions that sound like they decided against. Accessing the config by coalesce the environment variables is status quo right now for all the other configs so I actually think its fine to add this one with the same behavior. Resolving the poetry.lock conflicts seems like the only thing required right now but I'm not exactly sure what the best way is to do that on a fork branch.


Also it turns out I was getting `Required argument `s3_staging_dir` or `work_group` not found.` trying to run this on main because [s3_staging_dir](https://github.com/MeltanoLabs/target-athena/blob/f8cd5b587825bdb8e8a491e2765a4a32f6c2ddee/target_athena/athena.py#L32) is an available config option that isnt advertised. When I switched to the working group branch it was able to use the default primary working group and succeed, so this makes that bug less relevant.

**Pull request merged**